executable/optionrom.ksy

```diff
@@ -0,0 +1,474 @@
+meta:
+  id: optrom
```

Comment:
> Firmware is executable code, right?

Not always. Firmware is actually closer to a file system, in respect that it stores both executable and non-executable parts, structured in some sort of way that bootloader would be able to locate them. To be frank, when it *is* just executable code, it usually means that Kaitai Struct has nothing to do with it.

In case of option ROMs, as I see, .ksy describes tons of tables which are *not* executable code, but used by executable code to configure hardware stuff.

> Everything is a computer, why do we discriminate them against x86?

Absolutely no discrimination. I propose the following rule of thumb:

* Some complex binary data (or its parts) that's normally flashed into some ROM chip and there's no user-space tools to access it as a file system → `firmware` (note that "firmware" might it turn contain references to filesystems, and it's totally ok)
* Small, non-container binary data, usually flashed on some sort of dedicated hardware part as EEPROM that are read by other components of the system for configuration purposes (i.e. EDID, SPD, NFC tags, PCI & SCSI configuration pages, etc) → `hardware`
* Container for (arbitrary) data structures which is implicity named "file system" and/or there are tools for some OSes which allow us to mount it / use it as file system → `filesystem`
* Container files on regular "file systems", which are mostly used to store executable code + some data to aid loading it in regular operating systems → `executable`

It doesn't matter if it's x86, ARM, Xtensa, or something exotic.

Any more opinions? Cc @koczkatamas @LogicAndTrick @tschoening?